merkledag/node.go

```diff
@@ -107,9 +110,17 @@ func (n *Node) AddNodeLinkClean(name string, that *Node) error {
 	if err != nil {
 		return err
 	}
-	lnk.Name = name
+	n.AddRawLink(name, lnk)
 
+	return nil
+}
+
+// AddRawLink adds a pre-provided link to this node
```

Comment:
I dont think he means providing in the dht-bitswap sense, i think he just means 'pre-existing'